File: content/lessons/old/2021L/04_enums/enums.c
#include <stdio.h>

enum shirt_size {
    small,
    medium,
    large,
    xlarge
};

void print_size(enum shirt_size size) {
    printf("my size is ");
    if (size == small) {
        printf("small");
    } else if (size == medium) {
        printf("medium");
    } else if (size == large) {
        printf("large");
    } else if (size == xlarge) {
        printf("xlarge");
    } else {
        printf("unknown");
    }
    printf("\n");
}

int main() {
    enum shirt_size my_size = medium;
    print_size(my_size);
}

File: content/lessons/old/2021L/04_enums/enums.rs
#![allow(unused_assignments)]
#![allow(unused_variables)]
#![allow(dead_code)]

#[derive(Debug)]
enum NamedSize {
    Small,
    Medium,
    Large,
    XL,
}

#[derive(Debug)]
enum ShirtSize {
    Named(NamedSize),
    Numeric(u32),
}

fn main() {
    println!(
        "Isn't it strange that some clothes' sizes are adjectives like {:?},",
        ShirtSize::Named(NamedSize::Small)
    );
    println!(
        "but sometimes they are numbers like {:?}?",
        ShirtSize::Numeric(42)
    );
}
